packages/react/src/xcss-prop/index.ts

```diff
@@ -27,19 +27,31 @@ type XCSSPseudos<
     : never;
 };
 
+type XCSSMediaQuery<
+  TAllowedProperties extends keyof CSSProperties,
+  TAllowedPseudos extends CSSPseudos,
+  TAllowedMediaQueries extends string,
+  TRequiredProperties extends { requiredProperties: TAllowedProperties },
+  TSchema
+> = {
+  [Q in `@media ${TAllowedMediaQueries}`]?:
+    | MarkAsRequired<
```

Comment:
Given this is a union, will it actually be required?  Doing my head in!

Just more or less worth a test, eg. you had a TODO on a test, so perhaps there?

Reply:
Which one are we talking about?

Reply:
Do we require all properties in `MarkAsRequired` if you have `XCSSPsuedos`?

Let's say `requiredProperties: 'color'`, I think this would pass:
```tsx
css({
  color: 'red',
  '@media (whatever)': {
    '&:hover': { color: 'red' },
  },
});
```
_or_ this would pass:
```tsx
css({
  color: 'red',
  '@media (whatever)': {
    color: 'red',
  },
});
```

Could of course load the repo up to validate, sorry, I didn't do that…

Reply:
Good q. I'll test when I swing back.